### DISCUSSION TITLE
Libdwarf version 0.3.4

### DISCUSSION THREAD

**Discussion by davea42** [davea42/libdwarf-code#81] [General]
Fixed a bug in test scripts so 'make check -j8' will work properly.
Fixed a bug where reading a PE (Windows) object could fail for certain section virtual size values.
Added initializers to two uninitialized local variables in dwarfdump source so a compiler warning will not kill a --enable-wall build.
Replaced the libdwarf.pdf content, it is now generated by doxygen and latex and has a proper table of contents and index.
Added support for the meson build system.
Added dwarfexample/showsectiongroups.c so one can see libdwarf's interpretation of section groups; making it simple to
use libdwarf when section groups are important (relevant for Split Dwarf and COMDAT sections).

<hr /><em>This discussion was created from the release <a href='https://github.com/davea42/libdwarf-code/releases/tag/v0.3.4'>Libdwarf version 0.3.4</a>.</em>

**Comment by davea42** [comment]
It's come to our attention that building on Windows using cmake fails to define -DLIBDWARF_BUILD in 
src/lib/libdwarf/CMakeLists.txt. So alter  that file around line 77 in the following way to fix this.
The configure and meson builds do this already.
```
-        target_compile_options(${target} PRIVATE ${DW_FWALL})
+    target_compile_options(${target} PRIVATE -DLIBDWARF_BUILD
+            ${DW_FWALL})
```
Thanks to Mario M. for letting us know of the issue.

**Comment by vtorri** [comment]
that kind of problem always convince me that cmake is not a good build system

**Comment by vtorri** [comment]
btw, you should add a CI integration file for cmake like there is for autotools and meson

also, look at appveyor for Windows compilation : https://www.appveyor.com/

**Comment by vtorri** [comment]
or with github actions. exemple:

https://github.com/ebassi/graphene/blob/master/.github/workflows/msys2.yml
